lib/request_handler/fieldsets_parser.rb

```diff
@@ -7,9 +7,10 @@ class FieldsetsParser
     def initialize(params:, allowed: {}, required: [])
       @params = params
       allowed.reject! { |_k, v| v == false }
-      allowed.each_value do |option|
+      allowed.each_pair do |_key, value|
         raise InternalArgumentError, allowed: 'must be a Schema or a Boolean' unless
-          RequestHandler.configuration.validation_engine.valid_schema?(option) || option.is_a?(TrueClass)
+          RequestHandler.configuration.validation_engine.valid_schema?(value) ||
```

Comment:
maybe in dry 1.0 you can finally have a better check? in dry < 1.0 it was unfortunately not possible because of a mixture of types and schemas being tested here